jwst/datamodels/model_base.py

```diff
@@ -471,6 +472,9 @@ def on_save(self, path=None):
         current_date.format = 'isot'
         self.meta.date = current_date.value
 
+        # Enforce model_type to be the actual type of model being saved.
+        self.meta.model_type = self.model_type
```

Comment:
Currently, `model_type` is not set for a base `DataModel`.  This changes that behavior.  I haven't quite worked through the ramifications of that.

The one thing this will change, is that if an old JWST FITS file without the DATAMODL keyword is read in and then written back out, it will now have the DATAMODL keyword.  Before this PR, datamodels would not modify FITS files without the DATAMODL keyword.

Reply:
Offline discussion: This was determined to be fine.